Library/ViewModels/CommentsViewModelTests.swift

```diff
@@ -111,365 +113,365 @@ internal final class CommentsViewModelTests: TestCase {
-        self.loadCommentsAndProjectIntoDataSourceProject.assertValues([project], "New project is emitted")
+          .assertValues([envelope.comments], "No new comments are emitted")
+        self.loadCommentsAndProjectIntoDataSourceProject
+          .assertValues([.template], "No new projects are emitted")
 
-        let updatedEnvelope = CommentsEnvelope.multipleCommentTemplate
+        self.scheduler.advance()
 
-        withEnvironment(apiService: MockService(fetchCommentsEnvelopeResult: .success(updatedEnvelope))) {
-          self.vm.inputs.refresh()
+        self.loadCommentsAndProjectIntoDataSourceComments
+          .assertValues([envelope.comments, updatedEnvelope.comments], "New comments are emitted")
+        self.loadCommentsAndProjectIntoDataSourceProject
+          .assertValues([.template, .template], "Same project is emitted again")
+      }
+    }
+  }
 
-          self.loadCommentsAndProjectIntoDataSourceComments
-            .assertValues([envelope.comments], "No new comments are emitted")
-          self.loadCommentsAndProjectIntoDataSourceProject
-            .assertValues([project], "No new projects are emitted")
+  func testProjectPagination_WhenLimitReached_CommentsAreUpdatedInDataSource() {
+    self.loadCommentsAndProjectIntoDataSourceComments.assertDidNotEmitValue()
+    self.loadCommentsAndProjectIntoDataSourceProject.assertDidNotEmitValue()
 
-          self.scheduler.advance()
+    let envelope = CommentsEnvelope.singleCommentTemplate
 
-          self.loadCommentsAndProjectIntoDataSourceComments
-            .assertValues([envelope.comments, updatedEnvelope.comments], "New comments are emitted")
-          self.loadCommentsAndProjectIntoDataSourceProject
-            .assertValues([project, project], "Same project is emitted again")
-        }
-      }
-    }
+    withEnvironment(apiService: MockService(fetchCommentsEnvelopeResult: .success(envelope))) {
+      self.vm.inputs.configureWith(project: .template, update: nil)
+      self.vm.inputs.viewDidLoad()
 
-    func testProjectPagination_WhenLimitReached_CommentsAreUpdatedInDataSource() {
       self.loadCommentsAndProjectIntoDataSourceComments.assertDidNotEmitValue()
       self.loadCommentsAndProjectIntoDataSourceProject.assertDidNotEmitValue()
+      self.isCommentsLoading.assertValues([true])
 
-      let envelope = CommentsEnvelope.singleCommentTemplate
-      let project = Project.template
+      self.scheduler.advance()
 
-      withEnvironment(apiService: MockService(fetchCommentsEnvelopeResult: .success(envelope))) {
-        self.vm.inputs.configureWith(project: project, update: nil)
-        self.vm.inputs.viewDidLoad()
+      self.loadCommentsAndProjectIntoDataSourceComments.assertValues(
+        [envelope.comments],
+        "A set of comments is emitted."
+      )
+      self.loadCommentsAndProjectIntoDataSourceProject.assertValues([.template])
+      self.isCommentsLoading.assertValues([true, false])
 
```

Comment:
May want to test `shouldShowIndicator` state here as well.

Reply:
There's a test for it, some few lines below.

Reply:
I guess the idea is to test the "before" and "after" states of this output, just like the other ones.